src/enclave/rpc_sessions.h

```diff
@@ -38,6 +39,43 @@ namespace enclave
     std::atomic<size_t> next_client_session_id =
       std::numeric_limits<size_t>::max() / 2;
 
+    class NoMoreSessionsEndpointImpl : public enclave::TLSEndpoint
+    {
+    public:
+      NoMoreSessionsEndpointImpl(
+        size_t session_id,
+        ringbuffer::AbstractWriterFactory& writer_factory,
+        std::unique_ptr<tls::Context> ctx) :
+        enclave::TLSEndpoint(session_id, writer_factory, std::move(ctx))
+      {}
+
+      void recv(const uint8_t* data, size_t size) override
+      {
+        recv_buffered(data, size);
+
+        if (get_status() == Status::ready)
+        {
+          // Send HTTP response describing soft session limit
+          auto http_response = http::Response(HTTP_STATUS_SERVICE_UNAVAILABLE);
+          http_response.set_header(
+            http::headers::CONTENT_TYPE, http::headervalues::contenttype::TEXT);
+          const auto response_body = fmt::format(
+            "Service is currently busy and unable to serve new connections");
+          http_response.set_body(
```

Comment:
Seems like we could and perhaps even should have a set_body(const std::string&)?

Reply:
I don't think this is worth adding. Accessing the raw `http::Response` like this is pretty rare, and its interface surface is currently pretty small. I don't think this conversion is too onerous here, or repeated elsewhere.